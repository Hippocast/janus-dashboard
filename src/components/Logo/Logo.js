import React from 'react';
import PropTypes from 'prop-types';

import block from '../../helpers/bem-cn';
import './Logo.css';

const b = block('j-logo');

const propTypes = {
    className: PropTypes.string,
};

const Logo = ({ className }) => (
    <span className={b.mix(className)()}></span>
);

Logo.propTypes = propTypes;

export default Logo;
