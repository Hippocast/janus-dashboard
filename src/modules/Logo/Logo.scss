@import 'index.scss';

.j-logo {
    display: block;
    background-image: url(../../assets/img/company-logo.svg);
    background-repeat: no-repeat;
    background-size: cover;
}
