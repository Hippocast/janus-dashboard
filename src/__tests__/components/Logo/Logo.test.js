import React from 'react';
import renderer from 'react-test-renderer';

import Logo from '../../../components/Logo/Logo';

describe('Logo component', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(
                <Logo />
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders with additional className if passed', () => {
        const props = {
            className: 'mock-class',
        };
        const tree = renderer
            .create(
                <Logo {...props} />
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
