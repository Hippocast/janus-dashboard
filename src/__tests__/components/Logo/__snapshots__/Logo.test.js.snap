// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component renders correctly 1`] = `
<span
  className="j-logo"
/>
`;

exports[`Logo component renders with additional className if passed 1`] = `
<span
  className="j-logo mock-class"
/>
`;
